### PR TITLE
Match current endpoints to spec

### PR DESCRIPTION
**Pull request by thullSL**
* Change last-contacted  to last-contact 
* Use threads & threads.items not correspondence

### CODE REVIEW

File: Comms/LastContactService.fs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿module Comms.LastContactService
+
+open Comms.ThreadService
+open Comms.Models
+open System
+open System.Collections.Generic
+open Serilog
+
+let threadToLastContactSummary(thread : ThreadDetail): LastContactSummary =
+    let lastCorrespondence = thread.Items |> Seq.maxBy (fun c -> c.Date)
```
Comment (lclarksl) [inline Comms/LastContactService.fs]: What happens if the thread is empty?

File: Comms/Models.fs [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
+    Message : string
+    }
+
+type ThreadDetail = {
+    Id : string
+    Items : seq<Correspondence>
+}
+
+[<DynamoDBTable("Thread")>]
+type ThreadEntity()  =
```
Comment (lclarksl) [inline Comms/Models.fs]: In the other F# repos, we've separated models and entities into their own files.

File: Comms/Models.fs [hunk 3]
```diff
@@ -1,10 +1,44 @@
...
 open Amazon.DynamoDBv2.DataModel
+open System
 
-[<DynamoDBTable("Correspondence")>]
-type CorrespondenceItem () =
-    member val From = "" with get, set
-    member val To = "" with get, set
+type Correspondence = {
+    SenderId : string
+    ReceiverId : string
```
Comment (lclarksl) [inline Comms/Models.fs]: Did we decide that we didn't need the sender/receiver ids or have I made that up?

File: Comms/Models.fs [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+    Id : string
+    Items : seq<Correspondence>
+}
+
+[<DynamoDBTable("Thread")>]
+type ThreadEntity()  = 
+    member val Id = "" with get, set
+    
+[<DynamoDBTable("CorrespondenceItem")>]
+type CorrespondenceEntity() =
```
Comment (lclarksl) [inline Comms/Models.fs]: CorrespondenceItemEntity? It is a bit long though...

File: Comms/Models.fs (hunk 3 above)
Comment (thullSL) [inline Comms/Models.fs]: swing and a miss :baseball: 

[According](https://github.com/BristechSRM/comms/blob/master/Comms/api.json#L124) to the spec away

File: Comms/ThreadService.fs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    { SenderId = entity.SenderId
+      ReceiverId = entity.ReceiverId
+      Type = entity.Type
+      SenderHandle = entity.SenderHandle
+      ReceiverHandle = entity.ReceiverHandle
+      Date = entity.Date
+      Message = entity.Message }
+
+let getEntityItems (id : string) : seq<Correspondence> = 
+    Log.Information("Contacting DynamoDB for correspondence with {id}", id)
```
Comment (dwybourn) [inline Comms/ThreadService.fs]: This log message isn't very helpful. I just ran it and got the log message "Contacting DynamoDB for correspondence with "2""

**Comment by dwybourn** [comment]
I just ran this and I got the following message returned 
"{"message":"An error has occurred.","exceptionMessage":"Object reference not set to an instance of an object","exceptionType":"System.NullReferenceException","stackTrace":""

File: Comms/LastContactService.fs (hunk 1 above)
Comment (thullSL) [inline Comms/LastContactService.fs]: Throws an exception of course :wink: 

What do we want returned if there isn't anything?

```
{
        "ThreadId" : "1",
        "Date" : "unknown",
        "SenderId" : "unknown",
        "ReceiverId" : "unknown"
    }
```

**Comment by thullSL** [comment]
The Exception is from me testing a thread with no items to answer Lauren's question. Works again now

**Comment by dwybourn** [comment]
I'd recommend adding some more examples to the database, with just one item in there it's hard to tell if the logic to calculate the last-contact is working correctly.

File: Comms/LastContactService.fs (hunk 1 above)
Comment (lclarksl) [inline Comms/LastContactService.fs]: Not sure, @ianlovell ?

Comment (thullSL) [inline Comms/LastContactService.fs]: Nevermind, filtering them out instead. See: 85ce98b

Comment (lclarksl) [inline Comms/LastContactService.fs]: Do we care about the difference here between a thread not existing and a thread not having any correspondence items?

Comment (thullSL) [inline Comms/LastContactService.fs]: Not sure I understand. If a thread didn't exist we wouldn't know about it here.

**Comment by dwybourn** [comment]
I'm getting the same exception again, even with two items in the database.

**Comment by thullSL** [comment]
Have you pulled and rebuilt cuz I think I've fixed that

File: Comms/LastContactService.fs (hunk 1 above)
Comment (ianlovell) [inline Comms/LastContactService.fs]: Does it return an empty collection if noone is contacted?

**Comment by dwybourn** [comment]
I’ve pull and rebuilt and no longer get the exception message. However I’m only getting one item in the response, but there are two threads in the database.

**Comment by thullSL** [comment]
One of the threads doesn't have any correspondence yet so I could test whether a thread without correspondence throws that exception

File: Comms/LastContactService.fs (hunk 1 above)
Comment (thullSL) [inline Comms/LastContactService.fs]: Yes

Comment (ianlovell) [inline Comms/LastContactService.fs]: Also, are the JSON response body properties in camelCase?

**Comment by thullSL** [comment]
I've added three threads to the db for now. 
* one with no items
* one with two
* one with three

the end point seems to be functioning correctly with this data.

**Comment by dwybourn** [comment]
:shipit:

File: Comms/LastContactService.fs (hunk 1 above)
Comment (thullSL) [inline Comms/LastContactService.fs]: yes

**Comment by ianlovell** [comment]
:shipit:

**Pull request merged**